### DISCUSSION TITLE
Error in adonis - right-hand-side of formula has no usable terms

### DISCUSSION THREAD

**Discussion by rbfontana** [vegandevs/vegan#518] [Q&A]
Hi, 

I'm trying to run a phylogenetic eigenvector regression and selection (PVR.adonis - daee) that uses adonis to this, but for a reason, that I don't know, I'm having an error with adonis.  

Error message: "Error in vegan::adonis(traits ~ x[, include.temp, drop = FALSE], permutations = 1) : 
  right-hand-side of formula has no usable terms"

Does anyone have advice?

**Comment by jarioksa** [comment]
You should use a right-hand side that has usable terms. 

You should not use formula interface like that. The canonical way is:
```r
## adonis is deprecated: use adonis2 instead
adonis2(traits ~ used.term, data=x) # replace used.term with the name of your variable in x data frame
```
Your way works, if defined correctly, but errors are more easily made with your interface. For instance you may refer to a non-existing column. What do you see if you issue:
```r
x[, include.temp, drop = FALSE]
```
